src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -204,11 +206,31 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryMetadataEntry metad
 
         private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, RequestContext requestContext)
         {
+            string regionToUse = requestContext.ServiceBundle.Config.AuthorityInfo.RegionToUse;
+
             if (s_region.IsNullOrEmpty())
             {
-                s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+                try
+                {
+                    s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+
+                    if (!regionToUse.IsNullOrEmpty())
+                    {
+                        requestContext.ApiEvent.IsValidUserProvidedRegion = s_region.Equals(regionToUse);
+                    }
+                }
+                catch (MsalServiceException e)
```

Comment:
>e [](start = 44, length = 1)

you are not using this, remove and simply throw instead

Reply:
This is to catch the exception if the region was not detected. If in this case we don't even have a user provided region we throw the error and let the fallback to global logic handle it later. In case we have a user provided region, we don't throw an error but use that region.